desk/app/contacts.hoon

```diff
@@ -403,115 +425,254 @@
+          ::
+          =/  groups=(set $>(%cult value))
+            ?~  rof  ~
+            =+  set=(~(ges cy con.rof) groups+%cult)
+            ?:  =(~ set)  ~
+            (need set)
+          [%self (to-self-edit p.act-0 groups)]
+          ::
+            %contact-action-1
+          !<(action vase)
+        ==
       ?-  -.act
         %anon  p-anon:pub
-        %edit  (p-edit:pub p.act)
+        %self  (p-self:pub p.act)
+        %page  (p-page:pub p.act q.act)
+        ::  if we spot someone who is not a peer,
+        ::  we meet them first
+        ::
+        %spot  =?  cor  !(~(has by peers) p.act)
+                 si-abet:si-meet:(sub p.act)
+               (p-spot:pub p.act q.act)
+        %edit  (p-edit:pub p.act q.act)
+        %wipe  (p-wipe:pub p.act)
         %meet  (s-many:sub p.act |=(s=_s-impl:sub si-meet:s))
-        %heed  (s-many:sub p.act |=(s=_s-impl:sub si-heed:s))
         %drop  (s-many:sub p.act |=(s=_s-impl:sub si-drop:s))
         %snub  (s-many:sub p.act |=(s=_s-impl:sub si-snub:s))
       ==
     ==
+  ::  +peek: scry
+  ::
+  ::  v0 scries
+  ::
+  ::  /x/all -> $rolodex:legacy
+  ::  /x/contact/her=@ -> $@(~ contact-0:legacy)
+  ::
+  ::  v1 scries
+  ::
+  ::  /x/v1/self -> $contact
+  ::  /x/v1/book -> $book
+  ::  /x/v1/book/her=@p -> $page
+  ::  /x/v1/book/id/cid=@uv -> $page
+  ::  /x/v1/all -> $directory
+  ::  /x/v1/contact/her=@p -> $contact
+  ::  /x/v1/peer/her=@p -> $contact
   ::
   ++  peek
     |=  pat=(pole knot)
     ^-  (unit (unit cage))
     ?+    pat  [~ ~]
+      ::
         [%x %all ~]
-      =/  lor=rolodex
-        ?:  |(?=(~ rof) ?=(~ con.rof))  rol
-        (~(put by rol) our.bowl rof ~)
-      ``contact-rolodex+!>(lor)
-    ::
+      =/  rol-0=rolodex:legacy
+        %-  ~(urn by peers)
+        |=  [who=ship far=foreign]
+        ^-  foreign-0:legacy
+        =/  mod=contact
+          ?~  page=(~(get by book) who)
+            ~
+          q.u.page
+        (to-foreign-0 (foreign-mod far mod))
+      =/  lor-0=rolodex:legacy
+        ?:  |(?=(~ rof) ?=(~ con.rof))  rol-0
+        (~(put by rol-0) our.bowl (to-profile-0 rof) ~)
+      ``contact-rolodex+!>(lor-0)
+      ::
         [%x %contact her=@ ~]
       ?~  who=`(unit @p)`(slaw %p her.pat)
         [~ ~]
-      =/  tac=?(~ contact)
-        ?:  =(our.bowl u.who)  ?~(rof ~ con.rof)
-        =+  (~(get by rol) u.who)
-        ?:  |(?=(~ -) ?=(~ for.u.-))  ~
-        con.for.u.-
+      =/  tac=?(~ contact-0:legacy)
+        ?:  =(our.bowl u.who)
+          ?~(rof ~ (to-contact-0 con.rof))
+        =+  far=(~(get by peers) u.who)
+        ?:  |(?=(~ far) ?=(~ for.u.far))  ~
+        (to-contact-0 con.for.u.far)
       ?~  tac  [~ ~]
-      ``contact+!>(`contact`tac)
+      ``contact+!>(`contact-0:legacy`tac)
+      ::
+        [%x %v1 %self ~]
+      ?~  rof  [~ ~]
+      ?~  con.rof  [~ ~]
+      ``contact-1+!>(`contact`con.rof)
+      ::
+        [%x %v1 %book ~]
+      ?~  book  [~ ~]
+      ``contact-book-0+!>(book)
```

Comment:
is this mark correct? seems like this one and a few others below don't have the right marks

Reply:
Offending marks have been fixed.